### PR TITLE
chore: use pnpm and sn-gh-workflows v2

### PR DESCRIPTION
**Pull request by haxxmaxx**

### CODE REVIEW

**Comment by enell** [comment]
Did you use `pnpm import` to create the pnpm lock file?

**Comment by haxxmaxx** [comment]
> Did you use `pnpm import` to create the pnpm lock file?

nope just install

**Review by niekvanstaveren — COMMENTED** [review]
Couple of more things we probably need:

- [x] api specifications path
- [x] Secret for API key
- [x] Secret for NPM token

Changes look okay

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ on:
 
 jobs:
   build:
-    uses: qlik-oss/sn-gh-workflows/.github/workflows/build.yaml@v1
+    uses: qlik-oss/sn-gh-workflows/.github/workflows/build.yaml@v2
     secrets: inherit
     with:
       release: ${{ inputs.release || false}}
```
Comment (niekvanstaveren) [inline .github/workflows/build.yml:23]:
```suggestion
      release: ${{ inputs.release || false}}
      api_specification_path: api-specifications/properties.json
```

**Review by enell — APPROVED** [review]

**Pull request merged**